package-res/resources/web/pentaho/type/list.js

```diff
@@ -256,70 +310,75 @@ define([
        *
        * If `index` is negative,
        * it means the position at that many elements from the end (`index' = count - index`).
        *
        * When the index is greater than or equal to the length of the list,
        * the element or elements are appended to the list.
        *
-       * @param {any|any[]} fragment Element or elements to add.
+       * @param {any|Array} fragment Element or elements to add.
        * @param {number} index The index at which to start inserting new elements.
-       * @param {?boolean} [silent=false] Indicates that no events should be emitted.
        */
-      insert: function(fragment, index, silent) {
-        this._set(fragment, /*add:*/true, /*update:*/true, /*remove:*/false, /*index:*/index, silent);
+      insert: function(fragment, index) {
+        return this._set(fragment, /*add:*/true, /*update:*/true, /*remove:*/false, /*move:*/false, /*index:*/index);
       },
 
       /**
        * Removes one or more elements from the list.
        *
        * Specified elements that are not present in the list are ignored.
        *
-       * @param {pentaho.type.Element|Array.<pentaho.type.Element>} fragment Element or elements to remove.
+       * @param {any|Array} fragment Element or elements to remove.
        */
       remove: function(fragment) {
-        this._remove(fragment);
+        return this._usingChangeset(function(changeset) {
+          changeset._remove(fragment);
+        });
       },
 
       /**
        * Removes one or more elements from the list,
        * given the start index and the number of elements to remove.
        *
-       * If `count` is less than `1`, nothing is removed.
        * If `count` is {@link Nully} or omitted, it defaults to `1`.
+       * If `count` is less than `1`, nothing is removed.
        * If `start` is not less than the number of elements in the list, nothing is removed.
        * If `start` is negative,
        * it means to start removing that many elements from the end (`start' = length - start`).
        *
-       * @param {number} start Index at which to start removing.
-       * @param {number} [count=1] Number of elements to remove.
-       * @param {?boolean} [silent=false] Indicates that no events should be emitted.
+       * @param {number} start The index at which to start removing.
+       * @param {number} [count=1] The number of elements to remove.
        */
-      removeAt: function(start, count, silent) {
-        this._removeAt(start, count, silent);
+      removeAt: function(start, count) {
+        return this._usingChangeset(function(changeset) {
+          changeset._removeAt(start, count);
+        });
       },
 
       /**
        * Sorts the elements of the list using the given comparer function.
        *
        * @param {function(pentaho.type.Element, pentaho.type.Element) : number} comparer The comparer function.
-       * @param {?boolean} [silent=false] Indicates that no events should be emitted.
```

Comment:
In this case, it will not matter because the line in question was removed.

Nevertheless, I'll explain because it sheds light on the rest of the work.
Originally, invoking the method "sort" should trigger an event: an event is "emitted" and all of its listeners are executed.
When that parameter is set to "true", the method acts silently, i.e. the event does not take place, and none of the listeners are executed.